rust/src/push/mod.rs

```diff
@@ -522,6 +526,28 @@ fn test_deserialize_unstable_msc3931_condition() {
     ));
 }
 
+#[test]
+fn test_deserialize_unstable_msc3952_user_condition() {
+    let json = r#"{"kind":"org.matrix.msc3952.is_user_mention"}"#;
+
+    let condition: Condition = serde_json::from_str(json).unwrap();
+    assert!(matches!(
+        condition,
+        Condition::Known(KnownCondition::IsUserMention)
+    ));
+}
```

Comment:
Would it be worth tests that the `Condition` instance also serialises correctly to that piece of json? (It doesn't seem to be the done thing in this file, so probably not?)

Reply:
Maybe? I kind of just copied the other tests around here which seem to do this.

Reply:
Ahh, okay. I interpreted it as "does serde do what I think it's going to do?". (I have sometimes written tests to that effect when playing with Pydantic...(